src/test/codewhisperer/util/crossFileContextUtil.test.ts

```diff
@@ -0,0 +1,85 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as assert from 'assert'
+import * as path from 'path'
+import * as semver from 'semver'
+import { getRelevantCrossFiles } from '../../../codewhisperer/util/supplementalContext/crossFileContextUtil'
+import { shuffleList, closeAllEditors, toFile, assertTabSize } from '../../testUtil'
+import { makeTemporaryToolkitFolder, tryRemoveFolder } from '../../../shared/filesystemUtilities'
+import { normalize } from '../../../shared/utilities/pathUtils'
+
+// TODO: make it a util function inside testUtil.ts
+let tempFolder: string
+
+async function openATextEditorWithText(fileText: string, fileName: string): Promise<vscode.TextEditor> {
+    const completeFilePath = path.join(tempFolder, fileName)
+    toFile(fileText, completeFilePath)
+
+    const textDocument = await vscode.workspace.openTextDocument(completeFilePath)
+
+    return await vscode.window.showTextDocument(textDocument, { preview: false })
+}
+
+describe('getRelevantFiles', async function () {
+    before(async function () {
+        this.timeout(600000)
+    })
+
+    beforeEach(async function () {
+        tempFolder = await makeTemporaryToolkitFolder()
+    })
+
+    afterEach(async function () {
+        await closeAllEditors()
+        await tryRemoveFolder(tempFolder)
```

Comment:
If that doesn't work, and maybe even if it does: try using `createTestWorkspaceFolder`. https://github.com/aws/aws-toolkit-vscode/blob/4c9713be84882cdad8728f364b02f8546a202b72/src/test/testUtil.ts#L61

Then you don't need to actively delete the folder, because the test infra will clean up the folders after all tests are done: https://github.com/aws/aws-toolkit-vscode/blob/4c9713be84882cdad8728f364b02f8546a202b72/src/test/globalSetup.test.ts#L58